### PR TITLE
Fix kl_div_loss docstring formula and two doc nits

### PR DESCRIPTION
**Pull request by Pablosinyores**
### Summary

Three documentation fixes in the pure-Python layer, one of which corrects a formula a user would copy and get wrong results from.

**`kl_div_loss` documented formula (main fix).** The `reduction == 'none'` snippet reads `mx.exp(targets) * (targets - inputs).sum(axis)`. In Python `.sum(axis)` binds tighter than `*`, so that expression reduces the difference first and then broadcast-multiplies by `exp(targets)` — a different value *and* shape than the implementation, which sums the elementwise product over `axis`:

```python
loss = mx.sum(mx.exp(targets) * (targets - inputs), axis)
```

Corrected the snippet to `(mx.exp(targets) * (targets - inputs)).sum(axis)`. Verified against the existing `test_kl_div_loss` values: the code returns `[0.0, 0.831777]` (shape `[2]`), the old snippet returns shape `[2, 2]`, the parenthesized snippet matches the code. Added an assertion pinning the documented formula to the implementation.

**`Dropout2d` channel-order acronym.** Docstring said `NWHC`/`WHC`; MLX is channels-last, so it should be `NHWC`/`HWC` (`Dropout3d` already documents `NDHWC`). No behavior change — the layer masks whole channels.

**`Optimizer.init` argument name.** Docstring listed the argument as `model`; the parameter is `parameters`.

### Testing

`python -m unittest test_losses.TestLosses.test_kl_div_loss` passes; `black` clean. (No local Metal build — relying on CI for the full suite.)

### CODE REVIEW

**Review by zcbenz — APPROVED** [review]
Thanks!
